src/overlays/actors/ovl_En_Tru/z_en_tru.c

```diff
@@ -31,91 +89,1189 @@ const ActorInit En_Tru_InitVars = {
+        this->actor.focus.pos.x = (this->actor.focus.pos.x / 10.0f) * 10.0f;
+        this->actor.focus.pos.y = ((this->actor.focus.pos.y + 10.0f) / 10.0f) * 10.0f;
+        this->actor.focus.pos.z = (this->actor.focus.pos.z / 10.0f) * 10.0f;
+        Math_Vec3s_Copy(&this->actor.focus.rot, &this->actor.world.rot);
+        Matrix_MultiplyVector3fByState(&D_80A8B3FC, &this->unk_1F8);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A873B8.s")
+    if (!(this->unk_34E & 0x200) && (limbIndex == 14)) {
+        *dList = NULL;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A87400.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A875AC.s")
+void EnTru_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTru* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A8777C.s")
+    if (limbIndex == 19) {
+        func_80A86BAC(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A87880.s")
+void func_80A886D4(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
+    EnTru* this = THIS;
+    s32 pad[3];
+    s32 sp2C;
+    s32 phi_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A87B48.s")
+    if (this->unk_34E & 0x10) {
+        phi_v1 = false;
+    } else {
+        phi_v1 = true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A87DC0.s")
+    if (this->unk_34E & 0x20) {
+        sp2C = true;
+    } else {
+        sp2C = false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A87FD0.s")
+    if (!phi_v1) {
+        sp2C = false;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A881E0.s")
+    if (limbIndex == 21) {
+        func_8013AD9C(this->unk_366, this->unk_368 + this->actor.shape.rot.y, &this->unk_1EC, &this->unk_204, phi_v1,
+                      sp2C);
+        Matrix_StatePop();
+        Matrix_InsertTranslation(this->unk_1EC.x, this->unk_1EC.y, this->unk_1EC.z, MTXMODE_NEW);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        if (sp2C) {
+            s16 oldZ = this->unk_204.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/EnTru_Init.s")
+            this->unk_204.z = this->unk_204.x;
+            this->unk_204.x = oldZ;
+        }
+        Matrix_RotateY(this->unk_204.y, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->unk_204.x, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(this->unk_204.z, MTXMODE_APPLY);
+        Matrix_StatePush();
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/EnTru_Destroy.s")
+void EnTru_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static UNK_PTR D_80A8B408[] = {
+        &D_06018FA0,
+        &D_060197A0,
+        &D_06019FA0,
+        &D_060197A0,
+    };
```

Comment:
Almost certainly textures, but you should check.

Reply:
I dunno how to find them, leaving it as-is.